custom_components/vzug/api/__init__.py

```diff
@@ -567,14 +567,23 @@ async def get_zh_mode(self, *, default_on_error: bool = False) -> int:
         )
         return data["value"]
 
-    async def get_eco_info(self, *, default_on_error: bool = False) -> EcoInfo:
-        return await self._command(
+    async def get_eco_info(self, *, default_on_error: bool = False) -> EcoInfo | None:
```

Comment:
If you want to return `None` here this causes a chain of type issues further up. For instance `AggState` needs to be updated.

An easier way to handle this would be to return an empty dictionary (i.e. `return EcoInfo()`) here.
Since the types are marked as `total=False` it means that none of the keys are required. Returning an empty dictionary would be the equivalent of returning `None`.